src/components/SelectableMainTable.tsx

```diff
@@ -120,13 +120,18 @@ const SelectableMainTable: FC<Props> = ({
       ...(row.columns ?? []),
     ];
 
-    row.className = classnames(row.className, {
+    const className = classnames(row.className, {
       "selected-row": isRowSelected,
       "processing-row": isRowProcessing,
     });
-    row.key = row.name;
+    const key = row.name;
 
-    return row;
+    return {
+      ...row,
+      columns,
+      className,
+      key,
+    };
```

Comment:
Isn't this a performance issue, because we are copying all the data into a new object on each render cycle?

Reply:
Yeah I agree it is less performant. I think then alternative approach here is to not append more columns to each row if it has the same lengh as the number of headers.

if (row.columns?.length !== headersWithCheckbox.length) {
    row.columns = [
    {
        content: (
        <CheckboxInput
            label={
            <div className="u-off-screen">Select {row.name ?? "row"}</div>
            }
            labelClassName="u-no-margin--bottom"
            checked={isRowSelected}
            onChange={toggleRow}
            disabled={isRowProcessing || !row.name}
        />
        ),
        role: "rowheader",
        className: "select",
    },
    ...(row.columns ?? []),
    ];
}

Reply:
on second thought, I think destructuring each row wouldn't be that expensive. Most of the data sits within `row.colums`, when we destructure `row`, columns isn't destructured but passed by reference. The code would be safer that way, since mutating data by reference is a common source of bugs that is difficult to track down.